grunt/publish-modules.js

```diff
@@ -0,0 +1,94 @@
+module.exports = function(grunt) {
+  var fs = require('fs');
+  var path = require('path');
+  var spawn = require('child_process').spawn;
+
+  function npmPublish(cwd) {
+    return new Promise(function(resolve, reject) {
+      if (grunt.option('dry-run')) {
+        grunt.log.write('spawn npm publish {cwd: ' + cwd +'}\n');
+        resolve();
+      }
+      else {
+        var child = spawn('npm', ['publish'], {
+          stdio: 'inherit',
+          cwd: path.resolve(cwd || ''),
+        });
+        child.on('error', reject);
+        child.on('exit', resolve);
+      }
+    });
+  }
+
+  function promisify(fn, ctx) {
+    return function() {
+      var args = [].slice.call(arguments);
+      return new Promise(function(resolve, reject) {
+        args.push(function(error, result) {
+          if (error) {
+            reject(error);
+          }
+          else if (arguments.length === 2) {
+            resolve(result);
+          }
+          else {
+            resolve([].slice.call(arguments, 1));
+          }
+        });
+        fn.apply(ctx, args);
+      });
+    };
+  }
+
+  grunt.registerTask('publish-modules', function() {
```

Comment:
I'm +1 on omitting from package.json for the reasons you cite, and I think `grunt publish` is fine; I would suggest adding a README section or similar documentation describing the command and its process, however, so that this doesn't become opaque mystery over time.